src/engine/strat_engine/blockdev.rs

```diff
@@ -231,7 +248,7 @@ impl BlockDev {
         }
     }
 
-    pub fn wipe_metadata(&mut self) -> EngineResult<()> {
+    pub fn wipe_metadata(&self) -> EngineResult<()> {
```

Comment:
I bet if you empty ```self.block_devs``` in ```StratPool::destroy()```, i.e., consume the elements instead of iterating through them it will pass the compiler.